mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,492 @@
+        checkpoint artifact will have a metadata associated with it.
+        The metadata contains the evaluation metrics computed on validation
+        data with that checkpoint along with the current epoch. If True, it
+        also marks the checkpoint version with the best evaluation metric with
+        a 'best' alias. You can choose the best checkpoint in the W&B Artifacts
+        UI using this. It depends on `EvalHook` whose priority is more
+        than MMDetWandbHook.
+
+    - Evaluation: At every evaluation interval, the `MMDetWandbHook` logs the
+        model prediction as interactive W&B Tables. The number of samples
+        logged is given by `num_eval_images`. Currently, the `MMDetWandbHook`
+        logs the predicted bounding boxes along with the ground truth at every
+        evaluation interval. This depends on the `EvalHook` whose priority is
+        more than `MMDetWandbHook`. Also note that the data is just logged once
+        and subsequent evaluation tables uses reference to the logged data
+        to save memory usage. Please refer to
+        https://docs.wandb.ai/guides/data-vis to learn more about W&B Tables.
+
+    For more details check out W&B's MMDetection docs:
+    https://docs.wandb.ai/guides/integrations/mmdetection
+
+    ```
+    Example:
+        log_config = dict(
+            interval=10,
+            hooks=[
+                dict(type='MMDetWandbHook',
+                     init_kwargs={
+                         'entity': WANDB_ENTITY,
+                         'project': WANDB_PROJECT_NAME
+                     },
+                     interval=50,
+                     log_checkpoint=True,
+                     log_checkpoint_metadata=True,
+                     num_eval_images=100,
+                     bbox_score_thr=0.3)
+            ])
+    ```
+
+    Args:
+        init_kwargs (dict): A dict passed to wandb.init to initialize
+            a W&B run. Please refer to https://docs.wandb.ai/ref/python/init
+            for possible key-value pairs.
+        interval (int): Logging interval (every k iterations).
+            Default 10.
+        log_checkpoint (bool): Save the checkpoint at every checkpoint interval
+            as W&B Artifacts. Use this for model versioning where each version
+            is a checkpoint.
+            Default: False
+        log_checkpoint_metadata (bool): Log the evaluation metrics computed
+            on the validation data with the checkpoint, along with current
+            epoch as a metadata to that checkpoint.
+            Default: True
+        num_eval_images (int): Number of validation images to be logged.
+            Default: 100
+        bbox_score_thr (float): Threshold for bounding box scores.
+    """
+
+    def __init__(self,
+                 init_kwargs=None,
+                 interval=50,
+                 log_checkpoint=False,
+                 log_checkpoint_metadata=False,
```

Comment:
I think `log_checkpoint_metadata` is not necessary anymore. I will remove it.

Reply:
OK